Source/Core/Actors/ActorExecutionContext.cs

```diff
@@ -1462,6 +1471,14 @@ protected override void Dispose(bool disposing)
                 {
                     this.NameValueToActorId.Clear();
                     this.ProgramCounterMap.Clear();
+                    foreach (var id in this.ActorIds)
+                    {
+                        // clear the runtime to save memory on any leaked ActorId objects being held onto
+                        // by client code.
+                        id.Bind(null);
```

Comment:
I am worried though this might cause side-effects if existing customer code? Ideally any such held references should be cleaned inside a `TestIterationDisposeAttribute` callback that the user defines in their test, and this would be a user bug, not a Coyote bug. We can never stop them from holding a reference even into the runtime itself (which they can get access via the the test entry point:

```
static IActorRuntime  HeldRuntimeRef;
[Microsoft.Coyote.SystematicTesting.Test]
public static void Test(IActorRuntime runtime)
{
  HeldRuntimeRef = runtime;
}

[Microsoft.Coyote.SystematicTesting.TestIterationDisposeAttribute]
public static void TestCleanup()
{
  HeldRuntimeRef  = null;
}
```

:)

So then we don't need to maintain another data structure inside the runtime, which is less complexity.